src/express/defaultRequest.ts

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable no-param-reassign */
+import type { PayloadRequest, PayloadRequestContext } from './types';
+
+/**
+ * This makes sure that payloadRequest.payloadContext always exists (is {}) and populates it with an optional default context.
+ * This function mutates directly to avoid copying memory. As payloadRequest is not a primitive, the scope of the mutation is not limited to this function but should also be reflected in the calling function.
+ */
+export function populateDefaultRequest(payloadRequest: PayloadRequest|undefined, context?: PayloadRequestContext|undefined) {
```

Comment:
Few things here:
1. The function name should match the file name.
2. Should we have the name reflect that it is handling context? `setRequestContext` or something? Are you anticipating that this function would handle more default `req` behavior in the future? I don't see that being likely anytime soon.
3. The logic seems to have a lot of complexity for what little it is actually doing, perhaps I'm missing something.

Taking a stab at rewriting this, is this any different from what you've done?
```ts
export function setRequestContext(
  req: PayloadRequest = { payloadContext: null } as PayloadRequest,
  context: PayloadRequestContext = {},
) {
  if (req.payloadContext) {
    req.payloadContext = {
      ...req.payloadContext,
      ...context,
    };
  } else {
    req.payloadContext = context;
  }
}
```

Reply:
> Are you anticipating that this function would handle more default req behavior in the future? 

Yep, exactly. I've changed it to setRequestContext though, as we can always modify in the future if we do need it in the future.

> Taking a stab at rewriting this, is this any different from what you've done?

I've incorporated your use of default values to make it simpler. I did keep the `if (Object.keys(req.payloadContext).length === 0 && req.payloadContext.constructor === Object) {` though as that'd be another scenario where spread can be avoided as it's slow